cpp/log/sqlite_db-inl.h

```diff
@@ -317,9 +328,27 @@ template <class Logged>
 int64_t SQLiteDB<Logged>::TreeSize() const {
   cert_trans::ScopedLatency latency(
       latency_by_op_ms.GetScopedLatency("tree_size"));
-  std::lock_guard<std::mutex> lock(lock_);
+  std::unique_lock<std::mutex> lock(lock_);
 
   CHECK_GE(tree_size_, 0);
+  sqlite::Statement statement(
```

Comment:
Hacked up a little test program that only signs a new tree head, with 100K entries in the database. On master, 20 to 45 seconds, with this PR, 12 to 25 seconds. I used `sync && sudo sh -c 'echo 3 >/proc/sys/vm/drop_caches'` to minimize caching effects.

I'm guessing that this lazy evaluation isn't *twice* as fast because there's some cache warming for the tree signing from having walked the database once, but even then, it's a good fraction of 2x faster.